clientd3d/d3drender_objects.c

```diff
@@ -1225,7 +1227,27 @@ void D3DRenderObjectsDraw(d3d_render_pool_new* pPool, room_type* room,
 	// drawdata[] may contain more than one entry with the same id.
 	// We need to keep track of those we've already processed to avoid duplicates.
 	std::unordered_set<int> processedIds;
-	// base objects
+
+	// Sorting drawdata based on the draw.id field, with a condition to place non-DrawObjectType items last
+	std::sort(drawdata, drawdata + MAX_ITEMS, [](const DrawItem& a, const DrawItem& b) {
+		// Compare the type field directly, which is a BYTE
+		if (a.type != DrawObjectType && b.type == DrawObjectType) {
+			return false;
+		}
+		if (a.type == DrawObjectType && b.type != DrawObjectType) {
+			return true;
+		}
+
+		// If both items are of the DrawObjectType, compare their draw.id
+		if (a.type == DrawObjectType && b.type == DrawObjectType) {
+			// Compare draw.id from the object union member
+			return a.u.object.object->draw.id < b.u.object.object->draw.id;
```

Comment:
Heads up about this comment just above:
drawdata[] may contain more than one entry with the same id.

So I think that means the sort is not stable when there are two entries with the same ID.